.snippets/code/store-and-retrieve-data/use-storagehub-sdk/retrieve-your-data/retrieve-data.ts

```diff
@@ -1,128 +1,39 @@
-  const address = account.address;
-
-  // Create a wallet client using the defined chain, account, and RPC URL
-  const walletClient: WalletClient = createWalletClient({
-    chain,
-    account,
-    transport: http('https://services.datahaven-testnet.network/testnet'),
-  });
-
-  // --- Polkadot.js API setup ---
-  const provider = new WsProvider(
-    'wss://services.datahaven-testnet.network/testnet'
-  );
-  const polkadotApi: ApiPromise = await ApiPromise.create({
-    provider,
-    typesBundle: types,
-    noInitWarn: true,
-  });
-
-  // --- Connect to MSP Client and authenticate account ---
-  const baseUrl = 'https://deo-dh-backend.testnet.datahaven-infra.network/';
-  const httpConfig: HttpClientConfig = { baseUrl: baseUrl };
-  // A temporary authentication token obtained after Sign-In with Ethereum (SIWE).
-  // If not yet authenticated, this will remain undefined and the client will operate in read-only mode.
-  // Authentication is required for file downloads.
-  let sessionToken: string | undefined = undefined;
-  const sessionProvider = async () =>
-    sessionToken
-      ? ({ token: sessionToken, user: { address: address } } as const)
-      : undefined;
-  const mspClient = await MspClient.connect(httpConfig, sessionProvider);
-
-  // Check MSP Health Status
-  const mspHealth: HealthStatus = await mspClient.info.getHealth();
-  console.log('MSP service health:', mspHealth);
-
-  // Trigger the SIWE (Sign-In with Ethereum) flow.
-  // This prompts the connected wallet to sign an EIP-4361 message,
-  // which the MSP backend verifies to issue a JWT session token
-  const siweSession = await mspClient.auth.SIWE(walletClient);
-  console.log('SIWE Session:', siweSession);
-  // Store the obtained session token for future authenticated requests
-  sessionToken = (siweSession as { token: string }).token;
-
-  // --8<-- [end:initialize-and-setup]
-
-  // --- Retrieve data logic ---
-  // --8<-- [start:download-file]
-  const fileKey = 'INSERT_FILE_KEY_AS_HEX';
-
-  // Must be authenticated to download files
-  const downloadResponse: DownloadResult = await mspClient.files.downloadFile(
-    fileKey
-  );
-
-  // Check if the download response was successful
-  if (downloadResponse.status !== 200) {
-    throw new Error(`Download failed with status: ${downloadResponse.status}`);
-  }
-  // --8<-- [end:download-file]
-
-  // --8<-- [start:save-download]
-  // Define the local path where the downloaded file will be saved
-  // Here it is resolved relative to the current module’s URL.
-  const downloadPath = new URL(
-    '../../files/filename_downloaded.jpeg', // make sure the file extension matches the original file
+  // --8<-- [start:init-setup]
+  const fileKeyHex = 'INSERT_FILE_KEY_AS_HEX';
+  const fileKey = polkadotApi.createType('H256', fileKeyHex); // Convert to H256 type if not already
+  const filePath = new URL(`./files/INSERT_FILENAME.png`, import.meta.url)
```

Comment:
```suggestion
  // Make sure the file extension matches the original file
  const filePath = new URL(`./files/INSERT_FILENAME.png`, import.meta.url)
```

Seems like this should be above the line it is related to? Pls double-check if this is the correct line